src/main/java/com/epam/parso/impl/CharDecompressor.java

```diff
@@ -61,18 +61,19 @@ private CharDecompressor() {
             int endOfFirstByte = page[offset + currentByteIndex] & 0x0F;
             int countOfBytesToCopy;
             switch (controlByte) {
-                case 0x00:
-                case 0x10:
-                case 0x20:
-                case 0x30:
+                case (byte) 0x30://intentional fall through
+                case (byte) 0x20://intentional fall through
+                case (byte) 0x10://intentional fall through
+                case (byte) 0x00:
                     if (currentByteIndex != length - 1) {
-                        countOfBytesToCopy = (page[offset + currentByteIndex + 1] & 0xFF) + 64 + endOfFirstByte * 256;
+                        countOfBytesToCopy = (page[offset + currentByteIndex + 1] & 0xFF) + 64
+                           + page[offset + currentByteIndex] * 256;
```

Comment:
Yes, this change looks probable. Do you have example files where it makes difference?